### DISCUSSION TITLE
Conditionally rendering components in a shared template

### DISCUSSION THREAD

**Discussion by endymion1818** [muxinc/media-chrome#799] [Q&A] [answered]
I have a template that I want to customise based on whether it's live or not. I've defined my template like this

```
import { MediaThemeElement } from "media-chrome/dist/media-theme-element";

const template = document.createElement("template");

template.innerHTML = `
    <media-controller>
    <slot name="media" slot="media"></slot>
    <media-control-bar>
        <media-play-button></media-play-button>
    </media-control-bar>
        {{username ?? 'Unknown user'}}
        {{ !islive && <media-seek-backward-button></media-seek-backward-button> }}
    </media-controller>
`;

class TinyTheme extends MediaThemeElement {
  static template = template;
}

if (!globalThis.customElements.get("tiny-theme")) {
  globalThis.customElements.define("tiny-theme", TinyTheme);
}

export default TinyTheme;

```

Then consumed it like this:

```
import "media-chrome";
import "./style.css";
import "./tiny-theme";

const username = "bobbytables";
const islive = false;

document.querySelector<HTMLDivElement>("#app")!.innerHTML = `
  <tiny-theme username="${username}" islive="${islive}">
    <video
      slot="media"
      src="https://stream.mux.com/A3VXy02VoUinw01pwyomEO3bHnG4P32xzV7u1j1FSzjNg/high.mp4"
    ></video>
  </tiny-theme>
`;
```

However the output is as follows:

<img width="658" alt="Screenshot 2024-02-01 at 13 14 03" src="https://github.com/muxinc/media-chrome/assets/2216344/d37a933e-8671-44c6-8cfc-9250d5eecf0b">

How would I be able to conditionally render that component? Would it be better to provide different templates for each use case?

[Here is a codeSandbox demonstrating the issue](https://codesandbox.io/p/devbox/kind-cookies-k7sjrg?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cls20y30p00073b6hdsj8v6y6%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cls20y30o00023b6h0ce7sfln%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cls20y30o00043b6hmfbn32lz%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cls20y30p00063b6hx7hydunj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cls20y30o00023b6h0ce7sfln%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls20y30o00013b6hctazlvyn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpnpm-lock.yaml%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cls38166000023b6ijqz7p71l%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A6%252C%2522startColumn%2522%253A27%252C%2522endLineNumber%2522%253A7%252C%2522endColumn%2522%253A25%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cls20y30o00023b6h0ce7sfln%2522%252C%2522activeTabId%2522%253A%2522cls38166000023b6ijqz7p71l%2522%257D%252C%2522cls20y30p00063b6hx7hydunj%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls20y30p00053b6hsg6t7vaf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cls20y30p00063b6hx7hydunj%2522%252C%2522activeTabId%2522%253A%2522cls20y30p00053b6hsg6t7vaf%2522%257D%252C%2522cls20y30o00043b6hmfbn32lz%2522%253A%257B%2522id%2522%253A%2522cls20y30o00043b6hmfbn32lz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls20y30o00033b6h8pc6yn9j%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cls20y30o00033b6h8pc6yn9j%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by luwes** [comment, marked as answer]
the theme syntax is intentionally made very simple and less featured than JS syntax for example so conditional && or || are not supported.

conditionals are done like so https://www.media-chrome.org/docs/en/themes#conditionals

got it working in a fork
https://codesandbox.io/p/devbox/media-chrome-theme-conditional-3dx4kt

**Reply by endymion1818** [reply]
Thanks @luwes 

I can't see your sandboxes for some reason but I'll work off the docs 👍

**Reply by luwes** [reply]
Sorry about that, I updated the sandbox 
https://codesandbox.io/p/devbox/media-chrome-theme-conditional-3dx4kt

**Reply by endymion1818** [reply]
Ahh great, didn't get that thing about conditionals to render nested templates, that's awesome 👍 

```
        <template if="!islive">
          <media-seek-backward-button></media-seek-backward-button>
        </template>
```
